File: LearnView.js
import React, { useState } from 'react';
import { View, Text, TouchableOpacity, StyleSheet, ScrollView, TextInput } from 'react-native';

const LearnView = () => {
    const [selectedModule, setSelectedModule] = useState(null); // Track which module is expanded
    const [isChatExpanded, setIsChatExpanded] = useState(false); // State for chatbot expansion

    const links = (module) => {
        switch (module) {
            case "Credit":
                return ["Credit Basics Video", "Credit Management Test", "Credit Score Tips"];
            case "Insurance":
                return ["Insurance 101 Video", "Insurance Knowledge Test", "Types of Insurance"];
            case "Loans":
                return ["Understanding Loans Video", "Loan Application Test", "Loan Repayment Tips"];
            case "Savings":
                return ["Savings Strategies Video", "Savings Challenge Test", "How to Save Effectively"];
            default:
                return [];
        }
    };

    const moduleView = (title) => (
        <View>
            <TouchableOpacity style={styles.moduleButton} onPress={() => {
                setSelectedModule(selectedModule === title ? null : title);
            }}>
                <Text style={styles.moduleButtonText}>{title}</Text>
            </TouchableOpacity>
            {selectedModule === title && (
                <View style={styles.linksContainer}>
                    {links(title).map((link, index) => (
                        <Text key={index} style={styles.linkText}>{link}</Text>
                    ))}
                </View>
            )}
        </View>
    );

    const chatbotModule = () => (
        <View style={styles.chatbotContainer}>
            {!isChatExpanded ? (
                <TouchableOpacity style={styles.chatButton} onPress={() => setIsChatExpanded(true)}>
                    <Text style={styles.chatButtonText}>💬 Chat with me!</Text>
                </TouchableOpacity>
            ) : (
                fullScreenChatView()
            )}
        </View>
    );

    const fullScreenChatView = () => (
        <View style={styles.fullChatContainer}>
            <View style={styles.chatHeader}>
                <Text style={styles.chatHeaderText}>WealthWise AI Chatbot</Text>
            </View>
            <ScrollView style={styles.chatArea}>
                <Text>User: What can I learn about savings?</Text>
                <Text>WealthWise AI: You can explore various tips and tricks to save effectively...</Text>
                {/* Add more chat messages here */}
            </ScrollView>
            <View style={styles.inputContainer}>
                <TextInput
                    style={styles.inputField}
                    placeholder="Ask me a question..."
                    placeholderTextColor="gray"
                />
                <TouchableOpacity style={styles.sendButton}>
                    <Text style={styles.sendButtonText}>Send</Text>
                </TouchableOpacity>
            </View>
        </View>
    );

    return (
        <View style={styles.container}>
            <View style={styles.header}>
                <Text style={styles.headerText}>Learn</Text>
            </View>
            <ScrollView contentContainerStyle={styles.scrollContainer}>
                {moduleView("Credit")}
                {moduleView("Insurance")}
                {moduleView("Loans")}
                {moduleView("Savings")}
                {chatbotModule()}
            </ScrollView>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'white',
    },
    header: {
        backgroundColor: 'orange',
        height: 50,
        justifyContent: 'center',
        alignItems: 'center',
    },
    headerText: {
        color: 'white',
        fontWeight: 'bold',
        fontSize: 20,
    },
    scrollContainer: {
        paddingHorizontal: 16,
        paddingBottom: 20,
    },
    moduleButton: {
        backgroundColor: 'orange',
        padding: 12,
        borderRadius: 8,
        marginVertical: 5,
        width: '100%',
    },
    moduleButtonText: {
        color: 'white',
        fontWeight: 'bold',
        fontSize: 24,
        textAlign: 'center',
    },
    linksContainer: {
        backgroundColor: 'lightgray',
        borderRadius: 8,
        padding: 8,
        marginBottom: 10,
    },
    linkText: {
        padding: 8,
        borderBottomWidth: 1,
        borderBottomColor: 'gray',
    },
    chatbotContainer: {
        marginTop: 20,
        alignItems: 'center',
    },
    chatButton: {
        backgroundColor: 'blue',
        padding: 16,
        borderRadius: 10,
    },
    chatButtonText: {
        color: 'white',
        fontSize: 20,
    },
    fullChatContainer: {
        width: '100%',
        flex: 1,
        backgroundColor: 'white',
        borderRadius: 10,
        padding: 16,
    },
    chatHeader: {
        marginBottom: 10,
    },
    chatHeaderText: {
        fontSize: 24,
        fontWeight: 'bold',
    },
    chatArea: {
        flex: 1,
        marginBottom: 10,
    },
    inputContainer: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    inputField: {
        flex: 1,
        borderWidth: 1,
        borderColor: 'gray',
        borderRadius: 10,
        padding: 10,
        marginRight: 10,
    },
    sendButton: {
        backgroundColor: 'orange',
        padding: 10,
        borderRadius: 10,
    },
    sendButtonText: {
        color: 'white',
    },
});

export default LearnView;

File: frontend/wealthwise_ai/LearnView.swift
//
//  LearnView.swift
//  wealthwise_ai
//
//  Created by Allison Brown on 9/28/24.
//


import SwiftUI

struct LearnView: View {
    @State private var isChatExpanded = false
    @State private var selectedModule: String? // To track which module is expanded

    var body: some View {
        VStack {
            // Orange strip at the top
            Rectangle()
                .fill(Color.orange)
                .frame(height: 50)
                .overlay(
                    Text("Learn")
                        .font(.headline)
                        .foregroundColor(.white)
                )
            
            // List of modules, each in its own row
            VStack(spacing: 10) {
                moduleView(title: "Credit")
                moduleView(title: "Insurance")
                moduleView(title: "Loans")
                moduleView(title: "Savings")
            }
            .padding(.horizontal)
            
            // Chatbot Module
            chatbotModule
            
            Spacer()
        }
        .background(Color.white)
        .navigationTitle("") // Remove title from navigation bar
    }
    
    // Function to create a module view with dropdown
    private func moduleView(title: String) -> some View {
        VStack {
            Button(action: {
                withAnimation {
                    if selectedModule == title {
                        selectedModule = nil // Collapse if already open
                    } else {
                        selectedModule = title // Expand the selected module
                    }
                }
            }) {
                Text(title)
                    .font(.system(size: 24, weight: .bold)) // Larger font size
                    .padding()
                    .frame(maxWidth: .infinity)
                    .background(Color.orange)
                    .foregroundColor(.white)
                    .cornerRadius(8)
            }
            .padding(.vertical, 5)

            // Dropdown content
            if selectedModule == title {
                VStack(spacing: 5) {
                    ForEach(links(for: title), id: \.self) { link in
                        Text(link)
                            .padding()
                            .frame(maxWidth: .infinity)
                            .background(Color.gray.opacity(0.2))
                            .cornerRadius(8)
                    }
                }
                .padding(.horizontal)
                .transition(.slide) // Transition effect for dropdown
            }
        }
    }
    
    // Function to get links based on the module title
    private func links(for module: String) -> [String] {
        switch module {
        case "Credit":
            return ["Credit Basics Video", "Credit Management Test", "Credit Score Tips"]
        case "Insurance":
            return ["Insurance 101 Video", "Insurance Knowledge Test", "Types of Insurance"]
        case "Loans":
            return ["Understanding Loans Video", "Loan Application Test", "Loan Repayment Tips"]
        case "Savings":
            return ["Savings Strategies Video", "Savings Challenge Test", "How to Save Effectively"]
        default:
            return []
        }
    }
    
    // Chatbot Module
    private var chatbotModule: some View {
        VStack {
            if !isChatExpanded {
                Button(action: {
                    withAnimation {
                        isChatExpanded.toggle()
                    }
                }) {
                    Text("💬 Chat with me!")
                        .font(.title)
                        .padding()
                        .background(Color.blue)
                        .foregroundColor(.white)
                        .cornerRadius(10)
                }
            } else {
                fullScreenChatView
                    .onTapGesture {
                        withAnimation {
                            isChatExpanded.toggle()
                        }
                    }
            }
        }
        .padding()
    }
    
    // Full screen chat view for chatbot
    private var fullScreenChatView: some View {
        VStack {
            HStack {
                Text("WealthWise AI Chatbot")
                    .font(.title)
                    .foregroundColor(.black)
                Spacer()
            }
            .padding()
            
            Divider()
            
            Spacer()
            
            // Chat conversation area
            ScrollView {
                VStack(alignment: .leading, spacing: 10) {
                    Text("User: What can I learn about savings?")
                    Text("WealthWise AI: You can explore various tips and tricks to save effectively...")
                    // Add more chat messages here
                }
                .padding()
            }
            
            // Message input field
            HStack {
                TextField("Ask me a question...", text: .constant(""))
                    .padding(10)
                    .background(Color.gray.opacity(0.2))
                    .cornerRadius(10)
                
                Button(action: {
                    // Handle sending message
                }) {
                    Text("Send")
                        .padding(.horizontal, 10)
                        .background(Color.orange)
                        .foregroundColor(.white)
                        .cornerRadius(10)
                }
            }
            .padding()
        }
        .ignoresSafeArea() // Make it cover the entire screen
    }
}

struct LearnView_Previews: PreviewProvider {
    static var previews: some View {
        LearnView()
    }
}

